File: dotnet/Crtm05/CoordinateSystem.cs
namespace Crtm05
{
    public abstract class CoordinateSystem
    {
        public double Latitude { get; set; }
        public double Longitude { get; set; }

        protected CoordinateSystem(double latitude, double longitude)
        {
            Latitude = latitude;
            Longitude = longitude;
        }
    }
}

File: typescript/Crtm05/coordinate-system.ts
export abstract class CoordinateSystem {
  protected constructor(private latitude: number, private longitude: number) {}

  public getLatitude(): number {
    return this.latitude;
  }

  public getLongitude(): number {
    return this.longitude;
  }

  public setLatitude(latitude: number) {
    this.latitude = latitude;
  }

  public setLongitude(longitude: number) {
    this.longitude = longitude;
  }
}
